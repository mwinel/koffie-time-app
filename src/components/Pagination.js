import React, { Component } from "react";
import { Link } from "react-router-dom";

class Pagination extends Component {
  render() {
    return (
      <nav aria-label="Page navigation example">
        <ul className="pagination justify-content-center py-4">
          <li className="page-item disabled">
            <Link to="/all" className="page-link" tabIndex="-1">
              Prev
            </Link>
          </li>
          <li className="page-item">
            <Link to="/all" className="page-link">
              1
            </Link>
          </li>
          <li className="page-item">
            <Link to="/all" className="page-link">
              2
            </Link>
          </li>
          <li className="page-item">
            <Link to="/all" className="page-link">
              Next
            </Link>
          </li>
        </ul>
      </nav>
    );
  }
}

export default Pagination;
