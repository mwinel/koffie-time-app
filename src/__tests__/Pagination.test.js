import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";
import pretty from "pretty";
import Pagination from "../components/Pagination";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders pagination component", () => {
  act(() => {
    render(
      <BrowserRouter>
        <Pagination />
      </BrowserRouter>,
      container
    );
  });
  expect(container.innerHTML).toBeDefined();
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
