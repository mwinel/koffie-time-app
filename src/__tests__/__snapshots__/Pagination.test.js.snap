// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pagination component 1`] = `
"<nav aria-label=\\"Page navigation example\\">
  <ul class=\\"pagination justify-content-center py-4\\">
    <li class=\\"page-item disabled\\"><a class=\\"page-link\\" tabindex=\\"-1\\" href=\\"/all\\">Prev</a></li>
    <li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"/all\\">1</a></li>
    <li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"/all\\">2</a></li>
    <li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"/all\\">Next</a></li>
  </ul>
</nav>"
`;
